models/cart-discount/src/cart-discount-value-gift-line-item/cart-discount-value-gift-line-item-draft/builder.spec.ts

```diff
@@ -74,11 +74,11 @@ describe('builder', () => {
             typeId: 'product',
           }),
           variantId: expect.any(Number),
-          supplyChannel: expect.objectContaining({
+          supplyChannelRef: expect.objectContaining({
```

Comment:
Refs usually do not exist in GraphQL. They're a REST API concept. I am not sure we should be adding them to this model as a result.

Reply:
@dogayuksel had a look at the GraphQL introspection schema

![CleanShot 2024-05-10 at 11 45 58@2x](https://github.com/commercetools/test-data/assets/1877073/396f5015-9318-4ab1-b30c-5ed05deaf539)

and then GraphiQL. I think these refs exist on the value of cart discounts for instance

![CleanShot 2024-05-10 at 11 49 24@2x](https://github.com/commercetools/test-data/assets/1877073/b88cccf6-6afc-4a61-a62c-f805f683138c)

Reply:
yes i understand that part but i have this shape on card discounts, without this change our tests gets flaky when using the card discount model and by luck we get one gift line items.
``` 

      supplyChannelRef {
        id
        typeId
        __typename
      }
      distributionChannelRef {
        id
        typeId
        __typename
      }
    }
```

 https://github.com/commercetools/merchant-center-frontend/blob/main/packages-application/application-discounts/src/components/cart-discounts/details/cart-discount-details-connector/cart-discount-details-connector.ctp.graphql#L52-L62

Reply:
but as @doga infers i will close this pr and investigate further. thanks both your your inputs

Reply:
@dogayuksel  @tdeekens  how Is recomended to proceed here, my idea was to pass this info to BE and check the docs.  i can by pass this on our app adding a value when creating a card discount but i think we lost the idea here about having random value.

Im open to recomendations.

Reply:
`...ref` versions should be added to the non-draft side. Those seem to be implemented off from the spec, at least that is what is see looking at the generated types.

Reply:
I would argue they only exist on the none draft type first of all. Moreover, we can then fix that here our pass values when we use the model to restrict randomness. I'd like to learn why the actually exist if we add them here. Maybe a BE engineer can help.

Reply:
I already asked our BE I will post here when I get more info, thanks for your help.

Reply:
what stops me from doing that is I will need to modify the types that come from the platform, https://github.com/commercetools/commercetools-sdk-typescript/blob/ae315373964a2daba2186a38b795f708b4dc4058/packages/platform-sdk/src/generated/models/cart-discount.ts#L481

we use them to generate, i can omit some fields and add the other but I want to confirm with BE if rest and graphql
 bring the same fields if they are different we will fix one part and break another.